### PR TITLE
add dockerfile to bin-dist

### PR DESCRIPTION
**Pull request by teemow**
To make it easier to create your own Docker image of Mayu we ship the Dockerfile in tarball now. 

So now you fetch a tarball release from github and then adapt the config and create your own Docker image of Mayu. 

The builder release workflow here was broken anyway. Now builder doesn't create a docker image anymore (because of `Dockerfile.dist`). Question is how we handle that in general in the future.

### CODE REVIEW

**Comment by teemow** [comment]
ping @JosephSalisbury @htr @smaftoul

**Comment by teemow** [comment]
This has become a bigger bag of things. 

* updated etcd, coreos and yochu versions. 
* a dockerfile that is easy to rebuild with config changes
* new structure within the dockerfile eg. `/etc/mayu/config.yaml`, `/var/lib/mayu` etc.
* stripped down the singlenic mode template (networkd restart is not necessary here anymore)
* fixed order of things in the cloud config to have a reliable and reproducable start of things in the right order.

**Comment by teemow** [comment]
ping @htr @hectorj2f

File: Dockerfile.dist [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+RUN apt-get update && \
+    apt-get install -y \
+    dnsmasq=${DNSMASQ_VERSION} \
+    git=${GIT_VERSION} && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir -p /etc/mayu /var/lib/mayu /usr/lib/mayu
+COPY mayu mayuctl /usr/bin/
+COPY tftproot /usr/lib/mayu/tftproot
+COPY images /usr/lib/mayu/images
```
Comment (hectorj2f) [inline Dockerfile.dist]: @teemow You miss here to copy from the righ directory `./bin-dist/* ` or simply to use the path `./bin-dist/tftproot` as an example

Comment (teemow) [inline Dockerfile.dist]: @hectorj2f the dockerfile moved into the bin-dist folder and we need to create the docker image from inside there. reason: you can fetch the released tarball and create your own docker image (with your Mayu config) from it.

Comment (hectorj2f) [inline Dockerfile.dist]: :+1:

**Comment by hectorj2f** [comment]
@teemow code wise LGTM, I restarted the Travis process, it hanged when running `docker pull golang:1.5`

**Pull request merged**